src/tiler/server.js

```diff
@@ -63,8 +90,44 @@ var interactivity = {
             stream_order = 5;
         }
 
-        return '(SELECT geom, stream_order FROM ' + tableName +
-          ' WHERE stream_order >= ' + stream_order + ') as q';
+        var caseSql = function(mapping, field) {
+            bounds = _.map(Object.keys(mapping), function(category) {
+                return 'WHEN ' + field + ' >= ' + mapping[category][0] + ' AND ' + field +' < ' + mapping[category][1] + ' THEN \'' + category +'\' ';
+            });
+
+            bounds[0] = 'CASE ' + bounds[0];
+            bounds.push('WHEN ' + field + ' > ' + mapping.L4[1] + ' THEN \'L5\' ');
+            bounds.push('ELSE \'NA\' ');
+            bounds.push('END');
+
+            return _.reduce(bounds, function(acc, x) { return acc + ' ' + x; });
```

Comment:
I believe you're reducing this array to join the string elements inside.  You can achieve the same much more simply with `bounds.join(' ')` and we can remove the `underscore` dependency.